File: 0814-49-2425-bitwise-xor-of-all-pairings/0814-49-2425-bitwise-xor-of-all-pairings.py
from functools import reduce
from operator import ixor

class Solution:
    def xorAllNums(self, nums1: list[int], nums2: list[int]) -> int:
        xor1 = reduce(ixor, nums1, 0) if len(nums2) % 2 else 0
        xor2 = reduce(ixor, nums2, 0) if len(nums1) % 2 else 0
        return xor1 ^ xor2

File: 0814-49-2425-bitwise-xor-of-all-pairings/0814-49-2425-bitwise-xor-of-all-pairings.cpp
class Solution {
public:
    int xorAllNums(vector<int>& nums1, vector<int>& nums2) {
        return (nums2.size()&1 ? reduce(nums1.begin(),nums1.end(),0,bit_xor<>()) : 0 ) ^ (nums1.size()&1 ? reduce(nums2.begin(),nums2.end(),0,bit_xor<>()) : 0 );
    }
};